types/express-jwt/index.d.ts

```diff
@@ -15,6 +15,14 @@ export = jwt;
 declare function jwt(options: jwt.Options): jwt.RequestHandler;
 declare namespace jwt {
     type secretType = string | Buffer;
+    /**
+     * The express-jwt library specifies the following ErrorCode values by default:
+     * - `"revoked_token"`
+     * - `"invalid_token"`
+     * - `"credentials_bad_scheme"`
+     * - `"credentials_bad_format"`
+     * - `"credentials_required"`
+     */
     type ErrorCode = string;
```

Comment:
If you keep the previous type and union it with string, I believe Typescript etc. will treat it just as a string, but IDEs like VS Code will suggest autocompleting these for better developer experience (while still allowing any strings)

Reply:
@domdomegg I added the `string` type to the union of string literals, and unfortunately, autocomplete wasn't triggered in VS Code.
```ts
    type ErrorCode =
        | 'revoked_token'
        | 'invalid_token'
        | 'credentials_bad_scheme'
        | 'credentials_bad_format'
        | 'credentials_required'
        | string;
```

I found a relevant discussion thread on the main TS repo of others describing the same issue (https://github.com/microsoft/TypeScript/issues/29729) and came across the following pattern which _would_ support autocomplete while also allowing any `string`:

```ts
    type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);

    type ErrorCode = LiteralUnion<
        'revoked_token' | 'invalid_token' | 'credentials_bad_scheme' | 'credentials_bad_format' | 'credentials_required'
    >;
```

autocomplete:
<img width="923" alt="Screen Shot 2021-12-05 at 11 31 17 AM" src="https://user-images.githubusercontent.com/5847738/144755227-9cf639e2-d997-4e22-831f-4b97ac077f86.png">

Thoughts?

Reply:
it's used here, I've used it e.g. for `markdown-it-emoji`:
[types/markdown-it-emoji](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/markdown-it-emoji/index.d.ts#L25)

Reply:
good to know! I'll push a commit with this approach